gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/traverser/B_O_S_SE_SL_Traverser.java

```diff
@@ -110,17 +112,22 @@ public void merge(final Traverser.Admin<?> other) {
 
     /////////////////
 
+    final boolean carriesUnmergeableSack() {
+        // hmmm... serialization in OLAP destroys the transient sideEffects
+        return null != this.sack && (null == this.sideEffects || null == this.sideEffects.getSackMerger());
+    }
+
     @Override
     public int hashCode() {
-        return this.t.hashCode() + this.future.hashCode() + this.loops;
+        return carriesUnmergeableSack() ? System.identityHashCode(this) : (super.hashCode() ^ this.loops);
+    }
+
+    protected final boolean equals(final B_O_S_SE_SL_Traverser other) {
```

Comment:
I'm a little scared of this "double equals()" definition. @spmallette -- do you have any thoughts on this matter? Note that there is an `equals(Object)` method like always, but then there is this added `protected final equals(SpecificClass)`. I have a feeling that this could get screwy for languages like Groovy. Perhaps there is a better way to do what is needed without overloading `equals()`?